daemon/k8s_watcher.go

```diff
@@ -1969,6 +1967,37 @@ func (d *Daemon) deleteK8sPodV1(pod *v1.Pod) {
 	}
 }
 
+// missingK8sPodV1 returns all pods considered missing, if the IP doesn't exist in
+// the cache or if there is an endpoint associated with that pod that doesn't
+// contain all labels from the pod.
+func (d *Daemon) missingK8sPodV1(m versioned.Map) versioned.Map {
+	missing := versioned.NewMap()
+	for k, v := range m {
+		pod := v.Data.(*v1.Pod)
+		_, exists := ipcache.IPIdentityCache.LookupByIP(pod.Status.PodIP)
+		// We can only consider that a pods is missing if it doesn't exist
+		// in the identity cache.
+		if !exists {
+			missing.Add(k, v)
+			continue
+		}
+		podNSName := k8sUtils.GetObjNamespaceName(&pod.ObjectMeta)
+
+		podEP := endpointmanager.LookupPodName(podNSName)
+		if podEP == nil {
+			continue
```

Comment:
I would think the pod is missing in this case. Why not?

Reply:
@jrajahalme I added the following comment
```
		// Only 1 endpoint in the whole cluster is managing this pod, if it's
		// not found is due:
		// - pod is not being managed by this node;
		// - pod is running on this node but there are no endpoints associated
		//   with it. This association pod to endpoint association occurs when
		//   the endpoint is created, by the pkg/workload.
```
let me know if it's clear.